nixos/modules/tasks/filesystems/bcachefs.nix

```diff
@@ -235,6 +235,10 @@ in
         system.fsPackages = [ cfg.package ];
         services.udev.packages = [ cfg.package ];
 
+        boot.extraModulePackages = lib.optionals (!config.boot.kernelPackages.bcachefs.meta.broken) [
+          config.boot.kernelPackages.bcachefs
+        ];
```

Comment:
Could this lead to accidentally falling back to the in-tree version?
e.g. when latest kernel is not compatible yet

Reply:
Well we *have* to do that for < 6.16 right now. Hopefully by the time we're shipping 6.18 there won't *be* an in-tree module enabled in our kconfigs.

Reply:
Maybe it would be better to just add an eval warning if it's not using the out-of-tree module? After all, that use case is effectively unmaintained now.

Reply:
Well we wouldn't be triggering the warning in CI. All our in-tree uses should be using the right things.

Reply:
Is this going in soon? I'm hoping to write up a general bcachefs announcement soon with migration instructions for everyone, NixOS is the last major distro we're waiting on